raiden/messages.py

```diff
@@ -1416,6 +1416,217 @@ def from_dict(cls, data):
         return expired_lock
 
 
+class SignedBlindedBalanceProof(SignedMessage):
+    """Message sub-field `onchain_balance_proof` for `RequestMonitoring`.
+    """
+
+    def __init__(
+            self,
+            channel_identifier: typing.ChannelID,
+            token_network_address: typing.TokenNetworkAddress,
+            nonce: typing.Nonce,
+            additional_hash: typing.AdditionalHash,
+            chain_id: typing.ChainID,
+            signature: typing.Signature,
+            balance_hash: typing.BalanceHash,
+            non_closing_signature=b'',
+    ) -> typing.SignedBlindedBalanceProof:
+
+        super().__init__()
+
+        self.channel_identifier = channel_identifier
+        self.token_network_address = token_network_address
+        self.nonce = nonce
+        self.additional_hash = additional_hash
+        self.chain_id = chain_id
+        self.balance_hash = balance_hash
+        self.signature = signature
+        self.non_closing_signature = non_closing_signature
+
+    @classmethod
+    def from_balance_proof_signed_state(
+            cls: typing.SignedBlindedBalanceProof,
+            balance_proof: BalanceProofSignedState,
+    ) -> typing.SignedBlindedBalanceProof:
+        assert isinstance(balance_proof, BalanceProofSignedState)
+        return cls(
+            channel_identifier=balance_proof.channel_identifier,
+            token_network_address=balance_proof.token_network_identifier,
+            nonce=balance_proof.nonce,
+            additional_hash=balance_proof.message_hash,
+            chain_id=balance_proof.chain_id,
+            signature=balance_proof.signature,
+            balance_hash=hash_balance_data(
+                balance_proof.transferred_amount,
+                balance_proof.locked_amount,
+                balance_proof.locksroot,
+            ),
+        )
+
+    def _data_to_sign(self):
+        packed = pack_balance_proof(
+            nonce=self.nonce,
+            balance_hash=self.balance_hash,
+            additional_hash=self.additional_hash,
+            channel_identifier=self.channel_identifier,
+            token_network_identifier=self.token_network_address,
+            chain_id=self.chain_id,
+        )
+        return packed
+
+    def sign(self, privkey):
+        """Important: we don't write the signature to `.signature`"""
```

Comment:
nitpick: Hmm this seem more like a comment to compare this `sign()` with the `sign()` of other messages in this module. It should not be in the function docstring imo.